packages/amplify-graphql-transformer-core/src/transformation/transform.ts

```diff
@@ -362,16 +363,18 @@ export class GraphQLTransform {
         },
       });
       try {
-        sandboxNode.run(overrideCode, path.join(this.overrideConfig!.overrideDir, 'build', 'override.js')).override(appsyncResourceObj);
+        sandboxNode.run(overrideCode, overrideFilePath).override(appsyncResourceObj);
       } catch (err) {
         throw new AmplifyError('InvalidOverrideError', {
           message: 'Executing overrides failed.',
           details: err.message,
           resolution: 'There may be runtime errors in your overrides file. If so, fix the errors and try again.',
         }, err);
       }
+      return appsyncResourceObj;
     }
-    return appsyncResourceObj;
+
+    return {};
```

Comment:
if there's no override file or override has not been requested by caller returning `{}` is consistent with behavior on main branch.